### PR TITLE
Support MacOS13 x64/arm64 Agents Packer Templates on EC2

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Support MacOS13 x64/arm64 Packer Building EC2 AMI

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4670

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @peterzhuamazon moving forward we dont need [macos-disk-resize.sh](https://github.com/opensearch-project/opensearch-ci/blob/main/packer/scripts/macos/macos-disk-resize.sh) anymore, should we move this to legacy ?

File: packer/files/macos/bashrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:$PATH
-export PATH=/opt/local/bin:$PATH
+export PATH=/opt/homebrew/bin:/opt/local/bin:$PATH
```
Comment (prudhvigodithi) [inline packer/files/macos/bashrc:2]: Since this `bashrc` works for legacy `jenkins-agent-macos12-x64.json` , should we move this to legacy folder as well?

Comment (peterzhuamazon) [reply packer/files/macos/bashrc:2]: This also works for the new macos13 that is why I keep it here.

Thanks.

**Comment by peterzhuamazon** [comment]
> Thanks @peterzhuamazon moving forward we dont need [macos-disk-resize.sh](https://github.com/opensearch-project/opensearch-ci/blob/main/packer/scripts/macos/macos-disk-resize.sh) anymore, should we move this to legacy ?

I will keep this for a bit as I am still not 100 percent sure the resizing is consistent.
At least for now I am seeing the space from df, need more testing.

Thanks.

**Comment by peterzhuamazon** [comment]
Mend scan unrelated.

**Pull request merged**